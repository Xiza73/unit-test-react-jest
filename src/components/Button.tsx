export interface ButtonProps {
  type: "button" | "submit" | "reset";
  color: "primary" | "secondary";
  text: string;
  onClick: () => void;
}

const colorOptions = {
  primary: "bg-blue-500 hover:bg-blue-700 text-white",
  secondary: "bg-gray-500 hover:bg-gray-700 text-white",
};

export const Button = ({ type, color, text, onClick }: ButtonProps) => {
  return (
    <button
      type={type}
      className={`text-white font-bold py-2 px-4 rounded ${colorOptions[color]}`}
      onClick={onClick}
    >
      {text}
    </button>
  );
};
