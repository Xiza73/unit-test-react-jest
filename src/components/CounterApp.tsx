import { useState } from "react";

export interface CounterAppProps {
  value?: number;
}

export const CounterApp = ({ value }: CounterAppProps) => {
  const [counter, setCounter] = useState(value || 0);

  const handleAdd = () => {
    setCounter(counter + 1);
  };

  const handleSubtract = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value || 0);
  };

  return (
    <>
      <h1>Counter App</h1>
      <h2>{counter}</h2>
      <button onClick={handleAdd}>+</button>
      <button onClick={handleReset}>Reset</button>
      <button onClick={handleSubtract}>-</button>
    </>
  );
};
