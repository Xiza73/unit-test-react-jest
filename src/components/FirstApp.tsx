export interface FirstAppProps {
  title: string;
  subtitle: string;
  name: string;
}

export const FirstApp = ({ title, subtitle, name }: FirstAppProps) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
};
