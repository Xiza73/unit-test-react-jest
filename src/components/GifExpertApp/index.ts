export * from './Browser';
export * from './List';
