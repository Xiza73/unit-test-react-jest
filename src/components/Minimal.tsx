export interface SeparatorProps {
  className?: string;
}

export const Separator = ({ className }: SeparatorProps) => (
  <hr className={`w-full my-4 ${className}`} />
);
