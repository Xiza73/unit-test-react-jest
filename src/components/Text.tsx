export interface TitleProps {
  text: string;
}

export const Title = ({ text }: TitleProps) => (
  <h1 className="text-4xl font-bold text-center text-gray-800">{text}</h1>
);
