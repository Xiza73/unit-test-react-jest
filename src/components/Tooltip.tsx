import { useState } from "react";

export interface DangerTooltipProps {
  message: string;
}

export const DangerTooltip = ({ message }: DangerTooltipProps) => {
  const [show, setShow] = useState(false);

  return (
    <div
      className={`absolute z-10 bottom-2 right-2 ml-3 p-3 w-56 bg-red-700 rounded-md items-center justify-center ${
        show ? "flex" : "hidden"
      }`}
    >
      <div className="flex-shrink-0">
        <svg
          className="h-5 w-5 text-gray-300"
          viewBox="0 0 20 20"
          fill="currentColor"
        >
          <path
            fillRule="evenodd"
            d="M10 18a8 8 0 100-16 8 8 0 000 16zm0-2a6 6 0 110-12 6 6 0 010 12zm-1-5a1 1 0 112 0v2a1 1 0 11-2 0v-2zm1-4a1 1 0 100-2 1 1 0 000 2z"
            clipRule="evenodd"
          />
        </svg>
      </div>
      <div className="ml-3 w-0 flex-1 pt-0.5">
        <p className="text-sm leading-5 font-medium text-gray-300">{message}</p>
      </div>
      <div className="ml-4 flex-shrink-0 flex mb-auto">
        <button
          className="inline-flex text-gray-300 focus:outline-none focus:text-gray-400 transition ease-in-out duration-150"
          aria-label="Dismiss"
        >
          <svg className="h-5 w-5" viewBox="0 0 20 20" fill="currentColor">
            <path
              fillRule="evenodd"
              d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
              clipRule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  );
};
