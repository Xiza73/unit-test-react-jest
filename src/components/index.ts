export * from './GifExpertApp';
export * from './Button';
export * from './CounterApp';
export * from './FirstApp';
export * from './Minimal';
export * from './Text';
export * from './Tooltip';
