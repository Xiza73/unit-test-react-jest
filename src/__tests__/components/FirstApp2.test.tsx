import { render, screen } from "@testing-library/react";
import { FirstApp } from "../../components/FirstApp";

describe("Tests in <FirstApp />", () => {
  const title = "Hello World";
  const subtitle = "This is a subtitle";
  const name = "Manuel";

  test("should match the snapshot", () => {
    const { container } = render(
      <FirstApp title={title} subtitle={subtitle} name={name} />
    );

    expect(container).toMatchSnapshot();
  });

  test("should show the message 'Hello World'", () => {
    render(<FirstApp title={title} subtitle={subtitle} name={name} />);
    // screen.debug();
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("should show the title in a h1 tag", () => {
    render(<FirstApp title={title} subtitle={subtitle} name={name} />);

    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("should show the subtitle sent by props", () => {
    render(<FirstApp title={title} subtitle={subtitle} name={name} />);

    expect(screen.getByText(subtitle)).toBeTruthy();

    // if there are more than one element: getAllByText
    // expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});
