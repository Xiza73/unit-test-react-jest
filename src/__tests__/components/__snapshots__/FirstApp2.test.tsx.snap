// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> should match the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello World
  </h1>
  <p>
    This is a subtitle
  </p>
  <p>
    Manuel
  </p>
</div>
`;
